### PR TITLE
Get rid of using static package links and add support of Java 11

### PR DESCRIPTION
**Pull request by kharkevich**
Close #40 
* Throw away static links

### CODE REVIEW

**Comment by vnovichek** [comment]
Please postpone merging to master until #42 and #41 come to develop as well

File: tasks/fetch/s3.yml [hunk 1]
```diff
@@ -1,24 +1,32 @@
...
-- name: "Downloaded artifact"
+- name: Set downloaded artifact vars
   set_fact:
-    oracle_artifact: "{{ download_path }}/{{ transport_s3_path|basename }}"
-    oracle_artifact_basename: "{{ transport_s3_path|basename }}"
+    oracle_artifact: '{{ download_path }}/{{ transport_s3_path|basename }}'
+    oracle_artifact_basename: '{{ transport_s3_path|basename }}'
 
-- name: "Getting java version variables"
+- name: 'Split artifact name'
```
Comment (ViachaslauKabak) [inline tasks/fetch/s3.yml]: Seems we do not need single quotes here

File: tasks/fetch/s3.yml [hunk 2]
```diff
@@ -1,24 +1,32 @@
...
-    java_package: "{{ oracle_artifact_basename.split('-')[0] }}"
-    java_major_version: "{{ oracle_artifact_basename.split('-')[1].split('u')[0] }}"
-    java_minor_version: "{{ oracle_artifact_basename.split('-')[1].split('u')[1] }}"
-    java_arch: "{{ oracle_artifact_basename.split('-')[3] }}"
+    parts: >-
+      {{ oracle_artifact_basename |
+      regex_findall('^(.*j[dkre]{2})-([0-9]+)[u.]([0-9.]+)[-_]([a-z]+)-(x64|i586)') |
+      first | list }}
+
+- name: 'Set variables based on split'
```
Comment (ViachaslauKabak) [inline tasks/fetch/s3.yml]: Seems we do not need single quotes here

File: tasks/win_fetch/chocolatey.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+- name: Set java_binary_type = chocolatey
+  set_fact:
+    java_binary_type: chocolatey
+
+- name: Chocolatay will download artifact itself
```
Comment (ViachaslauKabak) [inline tasks/win_fetch/chocolatey.yml]: typo - chocolatey

File: tasks/win_fetch/chocolatey.yml [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  debug:
+    msg: 'Chocolatay will download artifact itself'
```
Comment (ViachaslauKabak) [inline tasks/win_fetch/chocolatey.yml]: typo - chocolatey

File: tasks/win_fetch/oracle-fallback.yml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    artifact_url_list: >-
+      {{ release_page['content'] |
+          regex_findall('https?://download.oracle.com/otn-pub/java/.*/' +
+                        java_package|string + '-(?:' + java_major_version|string +
+                        'u' + minor|string + '-|' + java_major_version|string +
+                        '.' + minor|string + '_)' +  os|string + '-' +
+                        java_arch|string + '(?!-demos).*' + extension|string) | list }}
+  register: artifact_url_fact
+
+- name: Show artifacts' urls found
```
Comment (ViachaslauKabak) [inline tasks/win_fetch/oracle-fallback.yml]: Do we need this debug?

File: tasks/win_fetch/oracle-fallback.yml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  set_fact:
+    artifact_checksum: >-
+      {{ release_page['content'] |
+        regex_findall('https?://download.oracle.com/otn-pub/java/.*/'
+          + java_package|string + '-(?:' + java_major_version|string + 'u'
+          + minor|string + '-|' + java_major_version|string + '.'
+          + minor|string + '_)' +  os|string + '-' + java_arch|string + '(?!-demos).*'
+          + extension|string + '.*?\",\"SHA256\":\"([a-fA-F\d]{64})') | last }}
+
+- name: Show artifacts' urls found
```
Comment (ViachaslauKabak) [inline tasks/win_fetch/oracle-fallback.yml]: Do we need this debug?

File: tasks/fetch/oracle-fallback.yml [hunk 7]
```diff
@@ -1,26 +1,83 @@
...
+    artifact_url_list: >-
+      {{ release_page['content'] |
+          regex_findall('https?://download.oracle.com/otn-pub/java/.*/' +
+                        java_package|string + '-(?:' + java_major_version|string +
+                        'u' + minor|string + '-|' + java_major_version|string +
+                        '.' + minor|string + '_)' +  os|string + '-' +
+                        java_arch|string + '(?!-demos).*' + extension|string) | list }}
+  register: artifact_url_fact
+
+- name: Show artifacts' urls found
```
[the remark this answers is not in the document]
Comment (tgadiev) [reply tasks/fetch/oracle-fallback.yml]: Yes, it's useful

File: tasks/win_fetch/oracle-fallback.yml (hunk 5 above)
Comment (tgadiev) [reply tasks/win_fetch/oracle-fallback.yml]: Yes, we need it

File: tasks/win_fetch/oracle-fallback.yml (hunk 6 above)
Comment (tgadiev) [reply tasks/win_fetch/oracle-fallback.yml]: Yes

File: tasks/fetch/oracle-fallback.yml [hunk 8]
```diff
@@ -1,26 +1,83 @@
...
   retries: 15
   delay: 5
-  delegate_to: "localhost"
-  connection: "local"
-  become: False
-  register: FILE_DOWNLOADED
+  until: "'OK' in file_downloaded.msg"
 
-- name: "Downloaded artifact"
+- name: Sed downloaded artifact variable
```
[the remark this answers is not in the document]
Comment (tgadiev) [reply tasks/fetch/oracle-fallback.yml]: Fixed

File: tasks/fetch/s3.yml (hunk 1 above)
Comment (tgadiev) [reply tasks/fetch/s3.yml]: Fixed

File: tasks/fetch/s3.yml (hunk 2 above)
Comment (tgadiev) [reply tasks/fetch/s3.yml]: Fixed

File: tasks/win_fetch/chocolatey.yml (hunk 3 above)
Comment (tgadiev) [reply tasks/win_fetch/chocolatey.yml]: Fixed

File: tasks/win_fetch/chocolatey.yml (hunk 4 above)
Comment (tgadiev) [reply tasks/win_fetch/chocolatey.yml]: Fixed

**Pull request merged**